src/tests/ftest/cart_fault_injection/pool_fault_inject.py

```diff
@@ -0,0 +1,231 @@
+        self.number_servers = len(self.hostlist_servers) - 1
+
+    def look_missed_request(self, cmd_stderr, msg=b"MS request error"):
+        """ Read dmg_stderr for the msg string
+        If found, the instance attribute self.failed_request is
+        increased by 1.
+        """
+        if msg in cmd_stderr:
+            self.failed_requests += 1
+
+    def create_and_write_into_container(self):
+        """ Create a container
+        Write objects into container
+        Close container
+        Look for missed requests shown by daos command.
+        """
+        # Get and write into container
+        self.container = self.get_container(self.pool,
+                                            namespace=self.container_namespace)
+        self.look_missed_request(self.container.daos.result.stderr)
+
+        # This is done through IOR
+        # No daos/dmg message to catch
+        for server in range(0, self.number_servers):
+            self.container.write_objects(server, obj_class=self.object_class)
+        # Close container to avoid problems when deleting
+        # The container and/or pool
+        self.container.close()
+        self.look_missed_request(self.container.daos.result.stderr)
+
+    def exclude_and_reintegrate(self):
+        """ Exclude a random server from the pool
+        Wait for rebuild
+        Reintegrate the server back
+        Wait for rebuild
+
+        Due to the nature of how wait_for_rebuild() is coded
+        we can only get the last dmg command output.
+        """
+        server_to_exclude = randint(0, len(self.hostlist_servers) - 1)
+        self.pool.exclude([server_to_exclude])
+        self.look_missed_request(self.pool.dmg.result.stderr)
+
+        self.pool.wait_for_rebuild(True)
+        self.look_missed_request(self.pool.dmg.result.stderr)
+        self.pool.wait_for_rebuild(False)
+        self.look_missed_request(self.pool.dmg.result.stderr)
+        self.pool.reintegrate(str(server_to_exclude))
+        self.pool.wait_for_rebuild(True)
+        self.look_missed_request(self.pool.dmg.result.stderr)
+        self.pool.wait_for_rebuild(False)
+        self.look_missed_request(self.pool.dmg.result.stderr)
+
+    def _clean(self):
+        """Destroy container and pool
+        Look for missed request on each executed command
+        """
+        self.container.destroy()
+        self.look_missed_request(
+            self.container.daos.result.stderr)
+        self.pool.destroy(force=1)
+        self.look_missed_request(self.pool.dmg.result.stderr)
+
+    def test_pool_services(self):
+        """ Test the following pool commands:
+                Create
+                Create container
+                Write container (not a pool command)
+                Exclude rank from pool
+                Reintegrate rank to pool
+                List
+                Destroy
+
+            All the request have fault injection of a probability of 5% which
```

Comment:
Is this comment out of date? Was this not increased to 100%.

Reply:
Partially, yes, some of the failures have a rate of 5% except of pool destroy, to avoid the issues within the teardown.